bin/init.bash

```diff
@@ -124,391 +124,38 @@ generate_default_config() {
-            ;;
-
-        upcloud)
-            storage_class=standard
-
-            yq4 -i '.networkPolicies.kubeSystem.upcloud.enabled = true' "${file}"
-            yq4 -i '.networkPolicies.kubeSystem.upcloud.ips = ["94.237.0.0/23"]' "${file}" # api.upcloud.com
-            yq4 -i '.networkPolicies.kubeSystem.upcloud.ports = [443]' "${file}"
-            ;;
-
-        exoscale | baremetal)
-            storage_class=rook-ceph-block
-            ;;
-
-        aws)
-            storage_class=ebs-gp2
-            ;;
-
-        azure)
-            storage_class=standard
-            ;;
-
-        none)
-            return
-            ;;
-    esac
-
-    replace_set_me "${file}" ".storageClasses.default" "\"${storage_class}\""
-}
-
 # Usage: set_object_storage <config-file>
-# baremetal support is experimental, keep as separate case until stable
 set_object_storage() {
     file="${1}"
     if [[ ! -f "${file}" ]]; then
         log_error "ERROR: invalid file - ${file}"
         exit 1
     fi
     case ${CK8S_CLOUD_PROVIDER} in
-        aws | exoscale)
-            object_storage_type="s3"
-            yq4 --inplace '.objectStorage.s3.region = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.s3.regionEndpoint = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.s3.forcePathStyle = false' "${file}"
-            ;;
-
-        citycloud | safespring | upcloud | elastx)
-            object_storage_type="s3"
-            yq4 --inplace '.objectStorage.s3.region = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.s3.regionEndpoint = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.s3.forcePathStyle = true' "${file}"
-            ;;
-
         azure)
-            object_storage_type="azure"
-            yq4 --inplace '.objectStorage.azure.resourceGroup = "set-me"' "${file}"
-            yq4 --inplace '.objectStorage.azure.storageAccountName = "set-me"' "${file}"
-            ;;
-
-        baremetal)
-            object_storage_type="none"
+            replace_set_me "${file}" ".objectStorage.azure.resourceGroup" "\"${CK8S_ENVIRONMENT_NAME}-storage\""
+            replace_set_me "${file}" ".objectStorage.azure.storageAccountName" "\"${CK8S_ENVIRONMENT_NAME}\""
             ;;
-
         none)
             return
             ;;
     esac
-
-    replace_set_me "${file}" ".objectStorage.type" "\"${object_storage_type}\""
-    if [ "$object_storage_type" == "azure" ]; then
-        replace_set_me "${file}" ".objectStorage.azure.resourceGroup" "\"${CK8S_ENVIRONMENT_NAME}-storage\""
-        replace_set_me "${file}" ".objectStorage.azure.storageAccountName" "\"${CK8S_ENVIRONMENT_NAME}\""
-    fi
 }
 
-# Usage: set_nginx_config <config-file>
-# baremetal support is experimental, keep as separate case until stable
-set_nginx_config() {
+# Usage: merge_provider_config <config-file>
+merge_provider_config() {
```

Comment:
Could we perhaps change this so we merge it in [together with the flavour](https://github.com/elastisys/compliantkubernetes-apps/blob/main/bin/init.bash#L121)?

```diff
# in generate_default_config()

  new_config=$(mktemp)
  append_trap "rm ${new_config}; chmod 444 ${default_config}" EXIT

+ local -a files
+ files=("${config_template_path}/config/${config_name}" "${config_template_path}/config/flavors/${CK8S_FLAVOR}/${config_name}")
+
+ if [[ -f "${config_template_path}/config/providers/${CK8S_CLOUD_PROVIDER}/${config_name}" ]]; then
+   files+=("${config_template_path}/config/providers/${CK8S_CLOUD_PROVIDER}/${config_name}")
+ fi

- # Change this to use one flavor for cloud provider and one for "size" (e.g. dev, growth, enterprise)
- envsubst < "${config_template_path}/config/${config_name}" | yq_merge -  "${config_template_path}/config/flavors/${CK8S_FLAVOR}/${config_name}" > "${new_config}"
+ yq_merge "${files[@]}" | envsubst > "${new_config}"

  chmod 644 "${default_config}"
  cat "${new_config}" > "${default_config}"
```
And with the envsubst as general case for Azure Blob Storage.